### PR TITLE
Add Feature to Disable "Video Streaming Backgrounded String" Feature

### PR DESCRIPTION
**Pull request by t-yoshii**
Add an option for app developer to disable the "Video Streaming Backgrounded String (SDL-0118)" feature.

### CODE REVIEW

**Comment by t-yoshii** [comment]
@theresalech The PR is ready for review.

File: proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Add Feature to Disable "Video Streaming Backgrounded String" Feature
+
+* Proposal: [SDL-NNNN](NNNN-filename.md)
```
Comment (theresalech) [inline proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md]: Please add filename here

File: proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Also, HMI can notify user that video is not available in another way, eg HU Native's dialog, other template etc, because HMI can detect that video streaming is suspended by OnVideoStreaming notification from SDL Core. So, on such HMI, it would be better to be able to delegate the notification method to HU side.
+
+Therefore, it will be better to offer an option to app developer to disable to show "Video Streaming Backgrounded String".
+
+
+## Proposed solution
+Add parameter `sendVideoBackgroundedString` to `SDLStreminMediaConfiguration` class to disable the "video streaming backgrounded string (SDL-0118)" feature. The default value of this parameter is `true` and "Video Streaming Backgrounded String" will be sent (Current behavior as of sdl_ios v6.2). If this parameter is set to `false`, `backgroundingPixelBuffer` will not be sent.
```
Comment (theresalech) [inline proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md]:
```suggestion
Add parameter `sendVideoBackgroundedString` to `SDLStreamingMediaConfiguration` class to disable the "video streaming backgrounded string (SDL-0118)" feature. The default value of this parameter is `true` and "Video Streaming Backgrounded String" will be sent (Current behavior as of sdl_ios v6.2). If this parameter is set to `false`, `backgroundingPixelBuffer` will not be sent.
```

File: proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Potential downsides
+
+App developer must be careful which solution to use, default "Video Streaming Backgrounded String" or other by checking HU make/model/trim in RAI response. However, as mentioned above, the default behavior will be *send the backgrounded string*, so, if app developer skip checking, nothing will be changed from sdl_ios v6.2.
```
Comment (theresalech) [inline proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md]:
```suggestion
App developer must be careful which solution to use, default "Video Streaming Backgrounded String" or other by checking HU make/model/trim in RAI response. However, as mentioned above, the default behavior will be *send the backgrounded string*, so, if app developer skips checking, nothing will be changed from sdl_ios v6.2.
```

File: proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Impact on existing code
+
+Minor API changes are required in `SDLStreminMediaConfiguration` and `SDLVideoLifecycleManager` in sdl_ios.
```
Comment (theresalech) [inline proposals/XXXX-add-feature-to-disable-video-streaming-backgrounded-string-feature.md]:
```suggestion
Minor API changes are required in `SDLStreamingMediaConfiguration` and `SDLVideoLifecycleManager` in sdl_ios.
```

**Comment by t-yoshii** [comment]
@theresalech ,
Thank you very much for your review. I updated the proposal based on your review.

**Pull request merged**